Framework/Algorithms/src/ReflectometryBackgroundSubtraction.cpp

```diff
@@ -133,39 +135,73 @@ void ReflectometryBackgroundSubtraction::calculatePolynomialBackground(
   poly->setProperty("XRanges", spectrumRanges);
   poly->setProperty("CostFunction", getPropertyValue("CostFunction"));
   poly->execute();
-  API::MatrixWorkspace_sptr bgd = poly->getProperty("OutputWorkspace");
+  MatrixWorkspace_sptr bgd = poly->getProperty("OutputWorkspace");
 
   // the background must be transposed again to get it in the same form as the
   // input workspace
   transpose->setProperty("InputWorkspace", bgd);
   transpose->execute();
-  API::MatrixWorkspace_sptr outputWS =
-      transpose->getProperty("OutputWorkspace");
+  MatrixWorkspace_sptr outputWS = transpose->getProperty("OutputWorkspace");
 
   outputWS = minus(inputWS, outputWS);
 
   setProperty("OutputWorkspace", outputWS);
 }
 
+/** Calculates the background by finding an average of the number of pixels each
+ * side of the peak. This is done using the python LRSubtractAverageBackground.
+ * The background is then subtracted from the input workspace.
+ *
+ * @param inputWS :: the input workspace
+ * @param indexList :: the ranges of the background region
+ */
+void ReflectometryBackgroundSubtraction::calculatePixelBackground(
+    MatrixWorkspace_sptr inputWS, const std::vector<double> &indexList) {
+
+  const std::vector<int> backgroundRange{static_cast<int>(indexList.front()),
+                                         static_cast<int>(indexList.back())};
+
+  auto outputWSName = getPropertyValue("OutputWorkspace");
+
+  MatrixWorkspace_sptr workspace = inputWS->clone();
+  AnalysisDataService::Instance().addOrReplace(outputWSName, workspace);
```

Comment:
It's generally not a good idea to make the algorithm dependent on the ADS; it should run also on the workspaces that are not stored there. Consider just setting the workspace property.

Reply:
I added it to the ADS originally because `LRSubtractAverageBackground` is a python algorithm and it was producing an error if I just set the workspace property but I have rewritten the algorithm slightly so I don't have to do that now.